### PR TITLE
feat: replace golint with revive

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

`golint` has been deprecated and the recommended replacement is `revive`

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/clusters/create.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (opts *CreateOpts) providerName() string {
 
 func (opts *CreateOpts) newReplicationSpec() atlas.ReplicationSpec {
 	var (
-		readOnlyNodes int64 = 0
+		readOnlyNodes int64
```
Comment (gssbzn) [inline internal/cli/atlas/clusters/create.go:157]: default is zero

**Review by threebee — APPROVED** [review]
LGTM!

File: internal/cli/atlas/clusters/create.go (hunk 1 above)
Comment (threebee) [reply internal/cli/atlas/clusters/create.go:157]: ah yes, I review similar change last week -> `var slaveDelay float64 = 0`. I'll keep it in mind going forward.

Comment (gssbzn) [reply internal/cli/atlas/clusters/create.go:157]: this is just the new linter is picky about it, I would not mind

**Pull request merged**